unit/testing-utils/use_catch.h

```diff
@@ -0,0 +1,37 @@
+/*******************************************************************\
```

Comment:
Nit: While I think this is a good approach to avoid patching the `catch.h` file, I do worry a bit that it's oh so easy for someone to directly include catch.h and we're then relying on code review eyeballs to spot it (or possibly CI if we have that running visual studio builds with warnings as errors enabled). Maybe thats enough, but I wonder if anything can be done with `module_dependencies.txt` to outlaw direct inclusion of `catch.h`? Alternatively, if we don't mind do a tiny modification to `catch.h`.... could we add a 3 line patch to `catch.h` that would `#error` if it's being directly included?

Reply:
@tautschnig I've just noticed you've pushed a commit that modifies `catch.hpp` to disable linting... if we're modifying that file for that purpose, could we also add the small modification to do something like ```
#ifndef INCLUDED_VIA_USE_CATCH_H
#error "Do not directly include catch.hpp - include use_catch.h instead"
#endif
```
at the top of the catch.hpp file (the same place you add the linting pragmas)? Then add the appropriate `#define INCLUDED_VIA_USE_CATCH_H` in your new use_catch.h file?

Reply:
Ah, apologies - I've just also noticed another new commit that moves `catch.hpp` and uses `module_depdencences.txt` to enforce this - apologies for the extra noise from me! :-)

Reply:
I'm wondering whether to add the extra safety net, though?

Reply:
I don't have a definite answer - probably the module_dependencies.txt is enough of a safety net, but it's "developer experience" isn't the best - i.e. it'll fail at CI time (not at build time on a local developers machine), and the cause of the failure isn't *entirely* clear (i.e. a developer would have to do some detective work to figure out why their code that builds fine locally, fails in CI, and what the correct course of action is). I'd view the additional patch to `catch.hpp` as a developer experience improvement, because it would fail on a developers local build, with a message that tells them what corrective action to take.

Reply:
Done as suggested, thanks!